Algorithm.Framework/Selection/QC500UniverseSelectionModel.cs

```diff
@@ -92,26 +98,28 @@ public override IEnumerable<Symbol> SelectCoarse(QCAlgorithm algorithm, IEnumera
         /// </summary>
         public override IEnumerable<Symbol> SelectFine(QCAlgorithm algorithm, IEnumerable<FineFundamental> fine)
         {
-            if (algorithm.Time.Month == _lastMonth)
-            {
-                return Universe.Unchanged;
-            }
-            _lastMonth = algorithm.Time.Month;
-
-            // The company's headquarter must in the U.S.
-            // The stock must be traded on either the NYSE or NASDAQ
-            // At least half a year since its initial public offering
-            // The stock's market cap must be greater than 500 million
             var filteredFine =
                 (from x in fine
                  where x.CompanyReference.CountryId == "USA" &&
                        (x.CompanyReference.PrimaryExchangeID == "NYS" || x.CompanyReference.PrimaryExchangeID == "NAS") &&
                        (algorithm.Time - x.SecurityReference.IPODate).Days > 180 &&
-                       x.EarningReports.BasicAverageShares.ThreeMonths *
-                       x.EarningReports.BasicEPS.TwelveMonths * x.ValuationRatios.PERatio > 500000000m
+                       x.MarketCap > 500000000m
                  select x).ToList();
 
-            var percent = _numberOfSymbolsFine / (double)filteredFine.Count;
+            var count = filteredFine.Count;
+
+            if (count == 0)
+            {
+                algorithm.Debug($@"QC500UniverseSelectionModel.SelectFine: Since no security has met the QC500 criteria,
+the current universe is unchanged.A new selection will be attempted on the next trading day.
+FineFundamental Count: {fine.Count()}");
```

Comment:
nit - space missing: `unchanged.A new`

Reply:
same here, all internal models should be silent. A comment is enough